Makefile

```diff
@@ -113,20 +120,30 @@ oci-build:
 oci-push:
 	$(OCI_RUNTIME) push ${IMG}
 
+legacy-manifest-dir:
+	@mkdir -p manifests/${CLUSTER_LOGGING_VERSION}
+
 # Generate bundle manifests and metadata, then validate generated files.
 .PHONY: bundle
-bundle: manifests $(KUSTOMIZE) $(OPERATOR_SDK)
+bundle: manifests legacy-manifest-dir $(KUSTOMIZE) $(OPERATOR_SDK)
 	$(OPERATOR_SDK) generate kustomize manifests -q
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	$(KUSTOMIZE) build config/manifests | $(OPERATOR_SDK) generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
 	$(OPERATOR_SDK) bundle validate ./bundle
+	@cp bundle/manifests/loki-operator.clusterserviceversion.yaml manifests/${CLUSTER_LOGGING_VERSION}/loki-operator.v${CLUSTER_LOGGING_VERSION}.clusterserviceversion.yaml
+	@cp bundle/manifests/loki.openshift.io_lokistacks.yaml manifests/${CLUSTER_LOGGING_VERSION}/loki.openshift.io_lokistacks.yaml
+	@cp bundle/manifests/loki-operator-controller-manager-metrics-service_v1_service.yaml manifests/${CLUSTER_LOGGING_VERSION}/loki-operator-controller-manager-metrics-service_v1_service.yaml
+	@cp bundle/manifests/loki-operator-manager-config_v1_configmap.yaml manifests/${CLUSTER_LOGGING_VERSION}/loki-operator-manager-config_v1_configmap.yaml
+	@cp bundle/manifests/loki-operator-metrics-monitor_monitoring.coreos.com_v1_servicemonitor.yaml manifests/${CLUSTER_LOGGING_VERSION}/loki-operator-metrics-monitor_monitoring.coreos.com_v1_servicemonitor.yaml
+	@cp bundle/manifests/loki-operator-metrics-reader_rbac.authorization.k8s.io_v1_clusterrole.yaml manifests/${CLUSTER_LOGGING_VERSION}/loki-operator-metrics-reader_rbac.authorization.k8s.io_v1_clusterrole.yaml
+	@cp bundle/manifests/loki-operator-prometheus_rbac.authorization.k8s.io_v1_role.yaml manifests/${CLUSTER_LOGGING_VERSION}/loki-operator-prometheus_rbac.authorization.k8s.io_v1_role.yaml
+	@cp bundle/manifests/loki-operator-prometheus_rbac.authorization.k8s.io_v1_rolebinding.yaml manifests/${CLUSTER_LOGGING_VERSION}/loki-operator-prometheus_rbac.authorization.k8s.io_v1_rolebinding.yaml
```

Comment:
Instead of hardcoding the `cp` commands for each individual file, we should make these into a make target that uses makefile idioms and caching. I have rewritten this rule like the following:

```
CLUSTER_LOGGING_VERSION ?= 5.1.preview.1

legacy-manifest-dir: manifests/$(CLUSTER_LOGGING_VERSION)/.keep
manifests/$(CLUSTER_LOGGING_VERSION)/.keep:
	@mkdir -p $(shell dirname $@) && touch $@

# legacy-manifest-files acts as an intermediary target by depending on all
# bundle/manifest files being copied to the manifests directory. The manifest
# yaml files are handled by the target immediately following this, where they
# simply cp source to target using a wildcard. Put simply, legacy-manifest-files
# maps to every yaml under bundle/manifest remapped to it's legacy path e.g.
#   'bundle/manifests/loki.openshift.io_lokistacks.yaml' -> 'manifests/5.1/loki.openshift.io_lokistacks.yaml'
# where manifests/5.1/loki.openshift.io_lokistacks.yaml is the target.
legacy-manifest-files: legacy-manifest-dir
legacy-manifest-files: legacy-csv
legacy-manifest-files: $(patsubst bundle/manifests/%.yaml, manifests/$(CLUSTER_LOGGING_VERSION)/%.yaml, $(wildcard bundle/manifests/*.yaml))
manifests/$(CLUSTER_LOGGING_VERSION)/%.yaml: bundle/manifests/%.yaml
	@cp -v bundle/manifests/$*.yaml $@

# The CSV does not work under the previous rule, because the
# CLUSTER_LOGGING_VERSION is part of the target filename so it requires it's
# own target
legacy-csv: manifests/$(CLUSTER_LOGGING_VERSION)/loki-operator.v$(CLUSTER_LOGGING_VERSION).clusterserviceversion.yaml
manifests/$(CLUSTER_LOGGING_VERSION)/loki-operator.v$(CLUSTER_LOGGING_VERSION).clusterserviceversion.yaml: bundle/manifests/loki-operator.clusterserviceversion.yaml
	@cp -v $^ $@

```

Notes: 

1. CLUSTER_LOGGING_VERSION should not have quotes because it invalidates the cache since Make does not understand quotes. It treats them as literal quotes and the targets will have embedded quotes in them, which will lead to misrepresentation of the target files. 
2. legacy-manifest-dir should be a _real_ target so that it does not get executed unless the target has changed. In GNU make a target _can technically_ be a directory, but in *nix based systems the timestamp of the directory is modified every time a file within the directory is updated. This invalidates the cache and breaks the chain. Instead, it's common to use a `.keep` file as the target whose timestamp is used for caching purposes. 
3. All yaml files under the bundle/manifests directory are created under a single, wildcard rule - except for the CSV. Make will handle the mapping of src -> dest. Because the CSV has the logging version in the filename, I created a separate target. The only caveat to this is that legacy-manifest-files will create the non-versioned CSV file. This _can_ be fix by using `$(filter-out ...)` to remove it from the src files, however, I don't think it's harmful to have `*.clusterserviceversion.yaml` and `*.<version>.clusterserviceversion.yaml`. If it is harmful then we can fix it.

Reply:
The last line of the `bundle` target needs to - temporarily - be `$(MAKE) legacy-manifest-files` because of the order of operations. I will have a followup PR to make the `bundle` target an _actual_ target like this one which will fix this issue. For now, we can have a temporary fix by adding `$(MAKE) legacy-manifest-files` to the end of the `bundle` target.

Reply:
Thank you so much for this explanation. I have always feared Makefile's text functions because I considered them harmful as per directories content change too often to spot misuse.